iot-hub/Samples/device/AndroidSample/things/src/main/java/android/iot/sdk/azure/microsoft/com/things/MainActivity.java

```diff
@@ -0,0 +1,251 @@
+                    client.closeNow();
+                    System.out.println("Shutting down...");
+                }
+                catch (Exception e)
+                {
+                    System.out.println("Exception while closing IoTHub connection: " + e.toString());
+                }
+            }
+        }).start();
+    }
+
+    private void start()
+    {
+        sendThread = new Thread(new Runnable() {
+            public void run()
+            {
+                try
+                {
+                    initClient();
+                    for(;;)
+                    {
+                        sendMessages();
+                        Thread.sleep(sendMessagesInterval);
+                    }
+                }
+                catch (InterruptedException e)
+                {
+                    System.out.println("Send Thread interrupted.");
+                    return;
+                }
+                catch (Exception e)
+                {
+                    System.out.println("Exception while opening IoTHub connection: " + e.toString());
+                }
+            }
+        });
+
+        sendThread.start();
+    }
+
+    private void sendMessages()
+    {
+        temperature = 20.0 + Math.random() * 10;
+        humidity = 30.0 + Math.random() * 20;
+        msgStr = "\"temperature\":" + String.format("%.2f",temperature) + ", \"humidity\":" + String.format("%.2f",humidity);
+        try
+        {
+            sendMessage = new Message(msgStr);
+            sendMessage.setProperty("temperatureAlert", temperature > 28 ? "true" : "false");
+            sendMessage.setMessageId(java.util.UUID.randomUUID().toString());
+            System.out.println(msgStr);
+            EventCallback eventCallback = new EventCallback();
+            client.sendEventAsync(sendMessage, eventCallback, 1);
+        }
+        catch (Exception e)
+        {
+            System.err.println("Exception while sending event: " + e.getMessage());
+        }
+    }
+
+    private void initClient() throws URISyntaxException, IOException
+    {
+        client = new DeviceClient(connString, protocol);
+
+        try
+        {
+            client.open();
+            MessageCallback callback = new MessageCallback();
+            client.setMessageCallback(callback, null);
+            client.subscribeToDeviceMethod(new SampleDeviceMethodCallback(), getApplicationContext(), new DeviceMethodStatusCallBack(), null);
+        }
+        catch (Exception e2)
+        {
+            System.err.println("Exception while opening IoTHub connection: " + e2.getMessage());
+            client.closeNow();
+            System.out.println("Shutting down...");
+        }
+    }
+
+    class EventCallback implements IotHubEventCallback
+    {
+        public void execute(IotHubStatusCode status, Object context)
+        {
+            Integer i = (Integer) context;
+            System.out.println("IoT Hub responded to message " + i.toString()
```

Comment:
check if context is instanceof Interger before typecasting

Reply:
fixed